assume/common/forecasts.py

```diff
@@ -174,27 +182,38 @@ def calc_forecast_if_needed(self):
         """
         Calculates the forecasts if they are not already calculated.
 
-        This method calculates additional forecasts if they do not already exist, including
-        "price_EOM" and "residual_load_forecast".
+        This method calculates price forecast and residual load forecast for available markets, if
+        thise don't already exist.
 
         Args:
         """
 
         cols = []
-        for pp in self.powerplants.index:
+        for pp in self.powerplants_units.index:
             col = f"availability_{pp}"
             if col not in self.forecasts.columns:
                 s = pd.Series(1, index=self.forecasts.index)
                 s.name = col
                 cols.append(s)
         cols.append(self.forecasts)
         self.forecasts = pd.concat(cols, axis=1).copy()
-        if "price_EOM" not in self.forecasts.columns:
-            self.forecasts["price_EOM"] = self.calculate_EOM_price_forecast()
-        if "residual_load_EOM" not in self.forecasts.columns:
-            self.forecasts[
-                "residual_load_EOM"
-            ] = self.calculate_residual_demand_forecast()
+
+        for market_id, config in self.market_configs.items():
+            if config["product_type"] != "energy":
+                self.logger.warning(
+                    f"Price forecast could be calculated for {market_id}. It can only be calculated for energy only markets for now"
+                )
+                continue
+
+            if f"price_{market_id}" not in self.forecasts.columns:
+                self.forecasts[
+                    f"price_{market_id}"
+                ] = self.calculate_EOM_price_forecast(market_id=market_id)
```

Comment:
calculate_market_price_forecast would suit the method name much better now right?

Reply:
yes true